InnovativeCars/force-app/main/default/classes/CarToShowroomHandler1.cls

```diff
@@ -0,0 +1,76 @@
+/**
+* @author Agnieszka Ząbkowicz
+* @date 06.2021
+*
+* @description handler for CarToShowroomTrigger
+*/
+public class CarToShowroomHandler1 {
+    
+    /**
+    * @author Agnieszka Ząbkowicz
+    * @date 06.2021
+    *
+    * @description method checking if car is already available in showroom 
+    * and adding error when appropriate
+    * @param newOffers list of offers objects to check
+    */
+    public static void checkIfCarIsAlreadyAvailable(List<Car_To_Showroom__c> newOffers) {
+        Set<Id> carIds = new Set<Id>();
+        for(Car_To_Showroom__c newOffer : newOffers) {
+            if(newOffer.Availability__c == 'Available at the showroom') {
+                carIds.add(newOffer.Car__c);
+            }
+        }
+        List<Car_To_Showroom__c> carToShowroomsList = new List <Car_To_Showroom__c>(
+            [SELECT Car__c, Showroom__c FROM Car_To_Showroom__c 
+             WHERE Availability__c LIKE 'Available at the showroom' 
+             AND Car__c IN :carIds]);
+        
+        for (Car_To_Showroom__c carToShowroom : newOffers){
+            if(carToShowroom.Availability__c == 'Available at the showroom'){
+                for(Car_To_Showroom__c existingCarToShowroom : carToShowroomsList){
+                    if(existingCarToShowroom.Car__c == carToShowroom.Car__c){
+                        if(existingCarToShowroom.Id != carToShowroom.ID){
+                        	carToShowroom.addError(
+                            	'The car is already available at showroom');
+                        }
+                        break;
+                    }
+                }
+            }
+        }
```

Comment:
for w forze z ifami to niewydajne rozwiazanie. Jezeli chodzi o ta czesc zadania moze warto zrobic albo set z juz istniejacymi idkiami pojazdow i pozniej tylko contain albo jakas mapa mogla by pomoc. Dodatkowo sprawdzasz nowo dodane z istniejacymi a co sie stanie jak dodam dla jednego pojazdu 2 oferty fizyczne?